DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces_overview/host-pipes/src/vector_add.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/sycl.hpp>  // sycl.hpp comes first to prevent compiler crash
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include "exception_handler.hpp"
+
+#define VECT_SIZE 256
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class IDSimpleVAddPipes;
+class IDPipeA;
+class IDPipeB;
+class IDPipeC;
+
+using PipeProps = decltype(sycl::ext::oneapi::experimental::properties(
+    sycl::ext::intel::experimental::ready_latency<0>));
+using InputPipeA =
+    sycl::ext::intel::experimental::pipe<IDPipeA, int, 0, PipeProps>;
+using InputPipeB =
+    sycl::ext::intel::experimental::pipe<IDPipeB, int, 0, PipeProps>;
+using OutputPipeC =
+    sycl::ext::intel::experimental::pipe<IDPipeC, int, 0, PipeProps>;
+
+struct SimpleVAddKernelPipes {
+  int len;
+
+  void operator()() const {
+    [[intel::initiation_interval(1)]]  // NO-FORMAT: Attribute
+    for (int idx = 0; idx < len; idx++) {
+      int a_val = InputPipeA::read();
+      int b_val = InputPipeB::read();
+      int sum = a_val + b_val;
+      OutputPipeC::write(sum);
+    }
+  }
+};
+
+int main() {
+  try {
+    // Use compile-time macros to select either:
+    //  - the FPGA emulator device (CPU emulation of the FPGA)
+    //  - the FPGA device (a real FPGA)
+    //  - the simulator device
+#if FPGA_SIMULATOR
+    auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+    auto selector = sycl::ext::intel::fpga_selector_v;
+#else  // #if FPGA_EMULATOR
+    auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+    // create the device queue
+    sycl::queue q(selector, fpga_tools::exception_handler);
+
+    // make sure the device supports USM host allocations
+    auto device = q.get_device();
+
+    std::cout << "Running on device: "
+              << device.get_info<sycl::info::device::name>().c_str()
+              << std::endl;
+
+    int count = VECT_SIZE;  // pass array size by value
+
+    // push data into pipes before invoking kernel
+    int *A = new int[count];
```

Comment:
snake_case for variables here and below
```suggestion
    int *a = new int[count];
```